### PR TITLE
Squash ranlib

### PR DESCRIPTION
**Pull request by aremmell**
Quiet down ranlib when building

### CODE REVIEW

**Comment by johnsonjh** [comment]
Let's merge readlink first so this is only the one char charge

**Comment by johnsonjh** [comment]
@aremmell looks like you might have an unrelated change in here too

File: Makefile [hunk 1]
```diff
@@ -152,7 +152,7 @@ shared: $(OBJ_SHARED)
 
 static: shared
 	ar -cr $(OUT_STATIC) $(OBJ_SHARED)
-	-($(RANLIB) "$(OUT_STATIC)" || true) > /dev/null 2>&1
+	-@($(RANLIB) "$(OUT_STATIC)" || true) > /dev/null 2>&1
```
Comment (johnsonjh) [inline Makefile:155]: Looks good

**Review by johnsonjh — APPROVED** [review]
👍🏻

**Comment by aremmell** [comment]
> 👍🏻

I wonder how perf is affected after another mutex lock (squelch spam) on every message.

**Comment by johnsonjh** [comment]
It's a little slower but by no means in acceptable -

```
        1000000 lines libsir(log file)...
        printf: 1000000 lines in 3.793sec (263671.7 lines/sec)
        libsir(stdout): 1000000 lines in 4.899sec (204108.2 lines/sec)
        libsir(log file): 1000000 lines in 1.920sec (520715.9 lines/sec)
        timer resolution: ~1nsec
```

**Comment by aremmell** [comment]
> It's a little slower but by no means unacceptable -
> 
> ```
>         1000000 lines libsir(log file)...
>         printf: 1000000 lines in 3.793sec (263671.7 lines/sec)
>         libsir(stdout): 1000000 lines in 4.899sec (204108.2 lines/sec)
>         libsir(log file): 1000000 lines in 1.920sec (520715.9 lines/sec)
>         timer resolution: ~1nsec
> ```

That seems like your terminal choked up; maybe restart it or do a `clear` -- look how much faster the file I/O was than stdio...

**Comment by johnsonjh** [comment]
Did `reset`:

```
        printf: 1000000 lines in 1.988sec (503113.7 lines/sec)
        libsir(stdout): 1000000 lines in 4.931sec (202780.5 lines/sec)
        libsir(log file): 1000000 lines in 1.844sec (542210.5 lines/sec)
        timer resolution: ~1nsec
```

**Comment by johnsonjh** [comment]
Not surprised at that, the SSD has lots of cache, the terminal has to actually draw

**Pull request merged**